src/sentry/profiles/consumers/process/factory.py

```diff
@@ -32,3 +36,11 @@ def create_with_partitions(
             health_checker=self.health_checker,
             next_step=next_step,
         )
+
+
+def is_sampled(headers) -> bool:
+    for k, v in headers:
+        if k == "sampled":
+            if isinstance(v, bytes):
+                return str(v, encoding="utf-8") == "true"
```

Comment:
`v.decode("utf-8")` is probably what you're looking for here

Reply:
@Zylphrex both works actually. Is there a substantial difference to opt for the decode method instead?

Reply:
https://docs.python.org/3/library/stdtypes.html#str

Don't think there's a huge difference but the docs mention

> Passing a [bytes](https://docs.python.org/3/library/stdtypes.html#bytes) object to [str()](https://docs.python.org/3/library/stdtypes.html#str) without the encoding or errors arguments falls under the first case of returning the informal string representation (see also the [-b](https://docs.python.org/3/using/cmdline.html#cmdoption-b) command-line option to Python).

which is why you need to specify the encoding here. This is just an uncommon usage of `str` that I don't think we have anywhere else. So the suggestion is more for style.